packages/bridging/src/providers/near-intents/NearIntentsBridgeProvider.ts

```diff
@@ -0,0 +1,370 @@
+    return sourceTokens
+  }
+
+  async getQuote(request: QuoteBridgeRequest): Promise<NearIntentsQuoteResult> {
+    const {
+      sellTokenAddress,
+      sellTokenChainId,
+      buyTokenAddress,
+      buyTokenChainId,
+      account,
+      amount,
+      receiver,
+      validFor,
+      owner,
+    } = request
+    const tokens = await this.api.getTokens()
+
+    const sellToken = getTokenByAddressAndChainId(tokens, sellTokenAddress, sellTokenChainId)
+    const buyToken = getTokenByAddressAndChainId(tokens, buyTokenAddress, buyTokenChainId)
+    if (!sellToken || !buyToken) throw new BridgeProviderQuoteError(BridgeQuoteErrors.NO_ROUTES)
+
+    const { quote, timestamp: isoDate } = await this.api.getQuote({
+      dry: false,
+      swapType: QuoteRequest.swapType.EXACT_INPUT,
+      slippageTolerance: request.slippageBps || 100, // fallback to 1%,
+      originAsset: sellToken.assetId,
+      depositType: QuoteRequest.depositType.ORIGIN_CHAIN,
+      destinationAsset: buyToken.assetId,
+      amount: amount.toString(),
+      refundTo: owner || account,
+      refundType: QuoteRequest.refundType.ORIGIN_CHAIN,
+      recipient: receiver || account,
+      recipientType: QuoteRequest.recipientType.DESTINATION_CHAIN,
+      deadline: calculateDeadline(validFor || 3600),
+    })
+
+    const slippage = (1 - Number(quote.amountOutUsd) / Number(quote.amountInUsd)) * 100
+
+    return {
+      isSell: request.kind === OrderKind.SELL,
+      depositAddress: quote.depositAddress as Hex,
+      quoteTimestamp: new Date(isoDate).getTime(),
+      expectedFillTimeSeconds: quote.timeEstimate,
+      limits: {
+        minDeposit: BigInt(quote.minAmountIn),
+        maxDeposit: BigInt(quote.amountIn),
+      },
+      fees: {
+        bridgeFee: BigInt(0),
+        destinationGasFee: BigInt(0),
+      },
+      amountsAndCosts: {
+        beforeFee: {
+          sellAmount: BigInt(quote.amountOut),
+          buyAmount: BigInt(quote.amountIn),
+        },
+        afterFee: {
+          sellAmount: BigInt(quote.amountOut),
+          buyAmount: BigInt(quote.amountIn),
+        },
+        afterSlippage: {
+          sellAmount: BigInt(quote.minAmountOut),
+          buyAmount: BigInt(quote.minAmountIn),
+        },
+        slippageBps: Math.trunc(slippage * 100),
+        costs: {
+          bridgingFee: {
+            feeBps: 0,
+            amountInSellCurrency: BigInt(0),
+            amountInBuyCurrency: BigInt(0),
+          },
+        },
+      },
+    }
+  }
+
+  async getUnsignedBridgeCall(request: QuoteBridgeRequest, quote: NearIntentsQuoteResult): Promise<EvmCall> {
+    const { account, sellTokenAddress, sellTokenChainId } = request
+    const { depositAddress } = quote
+
+    const cowShedAccount = this.cowShedSdk.getCowShedAccount(sellTokenChainId, account)
+
+    const adapter = getGlobalAdapter()
+    return createWeirollDelegateCall((planner) => {
+      const amount = planner.add(
+        createWeirollContract(
```

Comment:
If we need hooks, instead of weiroll we will need to build a wrapper contract. However, I don't think we need this hook. CoW Protocol allows to set the recipient, so we could makes the SDK instead to allow providers that don't require hooks. Your flow will be way simpler, and the orders more competitive because require less gas